skrub/_to_float32.py

```diff
@@ -0,0 +1,213 @@
+
+    >>> s = pd.Series([True, False], name='b')
+    >>> s
+    0     True
+    1    False
+    Name: b, dtype: bool
+    >>> to_float.fit_transform(s)
+    0    1.0
+    1    0.0
+    Name: b, dtype: float32
+
+    >>> s = pd.Series([True, None], name='b', dtype='boolean')
+    >>> s
+    0    True
+    1    <NA>
+    Name: b, dtype: boolean
+    >>> to_float.fit_transform(s)
+    0    1.0
+    1    NaN
+    Name: b, dtype: float32
+    >>> s = pd.Series([True, None], name='b')
+    >>> s
+    0    True
+    1    None
+    Name: b, dtype: object
+    >>> to_float.fit_transform(s)
+    0    1.0
+    1    NaN
+    Name: b, dtype: float32
+
+    float64 columns are converted to float32:
+
+    >>> s = pd.Series([1.1, 2.2])
+    >>> s
+    0    1.1
+    1    2.2
+    dtype: float64
+    >>> to_float.fit_transform(s)
+    0    1.1
+    1    2.2
+    dtype: float32
+
+    Float64Dtype and Float32Dtype are cast to ``np.float32``. We do this because
+    most scikit-learn estimators cannot handle ``pd.Float32Dtype`` correctly
+    yet, especially in the presence of missing values (represented by ``pd.NA``
+    in such columns).
+
+    >>> s = pd.Series([1.1, 2.2, None], dtype='Float32')
+    >>> s
+    0     1.1
+    1     2.2
+    2    <NA>
+    dtype: Float32
+    >>> to_float.fit_transform(s)
+    0    1.1
+    1    2.2
+    2    NaN
+    dtype: float32
+
+    Notice that ``pd.NA`` has been replaced by ``np.nan``.
+
+    Columns that cannot be cast to numbers are rejected:
+
+    >>> s = pd.Series(['1.1', '2.2', 'hello'], name='x')
+    >>> to_float.fit_transform(s)
+    Traceback (most recent call last):
+        ...
+    skrub._on_each_column.RejectColumn: Could not convert column 'x' to numbers.
+
+    Once a column has been accepted, all calls to ``transform`` will result in the
+    same output dtype. Values that fail to be converted become null values.
+
+    >>> to_float = ToFloat32().fit(pd.Series([1, 2]))
+    >>> to_float.transform(pd.Series(['3.3', 'hello']))
+    0    3.3
+    1    NaN
+    dtype: float32
+
+    Categorical and datetime columns are always rejected:
+
+    >>> s = pd.Series(['1.1', '2.2'], dtype='category', name='s')
+    >>> s
+    0    1.1
+    1    2.2
+    Name: s, dtype: category
+    Categories (2, object): ['1.1', '2.2']
+    >>> to_float.fit_transform(s)
+    Traceback (most recent call last):
+        ...
+    skrub._on_each_column.RejectColumn: Refusing to cast column 's' with dtype 'category' to numbers.
+    >>> to_float.fit_transform(pd.to_datetime(pd.Series(['2024-05-13'], name='s')))
+    Traceback (most recent call last):
+        ...
+    skrub._on_each_column.RejectColumn: Refusing to cast column 's' with dtype 'datetime64[ns]' to numbers.
+
+    float32 columns are passed through:
+
+    >>> s = pd.Series([1.1, None], dtype='float32')
+    >>> to_float.fit_transform(s) is s
+    True
+    """
+
+    def __init__(self, force_float32=True):
+        self.force_float32 = force_float32
```

Comment:
I missed it: where is it used ?

Reply:
nowhere, it lingered after I removed the option not to force float32 :D I'll remove it